packages/near-membrane-base/src/__tests__/environment.spec.ts

```diff
@@ -0,0 +1,149 @@
+import { VirtualEnvironment } from '../environment';
+
+describe('VirtualEnvironment', () => {
+    afterEach(() => {
+        jest.resetAllMocks();
+    });
+
+    it('throws when options bag is missing', () => {
+        expect.assertions(1);
+        expect(() => {
+            // @ts-ignore
+            // eslint-disable-next-line no-new
+            new VirtualEnvironment();
+        }).toThrow();
+    });
+    describe('VirtualEnvironment.prototype.remap', () => {
+        it('Skips index-like properties', () => {
+            expect.assertions(1);
+
+            const ve = new VirtualEnvironment({
+                blueGlobalThis: globalThis,
+                redGlobalThis: globalThis,
```

Comment:
The case of blue and red global this being the same global is one we haven't tested. Any way to not do this?

Reply:
In this case, it's not really important since what we're really after is the behavior of `.remap(...)`, which provide some testable objects to operate on. This mechanism is purely here to get us a `VirtualEnvironment` instance.

Reply:
right but remapping from blue-to-red is odd if both are the same.

Reply:
I'm sure that's very odd, but `blueGlobalThis` isn't actually used in the `VirtualEnvironment` constructor (it's not immediately clear to me why it's needed at all, but that's another question for another day). I'm just defining it here because TS demands that I do so. It's also not used in `remap` either 🤷